injector/__init__.py

```diff
@@ -389,6 +394,14 @@ class ImplicitBinding(Binding):
 _InstallableModuleType = Union[Callable[['Binder'], None], 'Module', Type['Module']]
 
 
+class UniqueBindings(UserDict[type, Binding]):
+    """A dictionary that raises an exception when trying to add duplicate bindings."""
+    def __setitem__(self, key: type, value: Binding) -> None:
+        if key in self.data:
+            raise NonUniqueBinding(key.__name__)
```

Comment:
I would like to see a more human readable error message here. Our other errors help the more novice users understad what is wrong.

Reply:
I completely agree! I will add a message that explains "Binding for 'X' already exists". Do you think that is detailed enough, or would you like to seeing some mention of the `unique` flag?

Reply:
I think the first message would be good enough, but that it would be even better to also mention the flag.